internal/app/service.go

```diff
@@ -742,29 +741,16 @@ func (svc *Service) BuildContext(correlationId string, contentType string) inter
 }
 
 // Publish pushes data to the MessageBus using configured topic
-func (svc *Service) Publish(data any) error {
-	messageClient := bootstrapContainer.MessagingClientFrom(svc.dic.Get)
-	if messageClient == nil {
-		return fmt.Errorf(messageBusDisabledErr)
-	}
-
-	triggertopic := svc.config.Trigger.PublishTopic
-	baseTopic := svc.config.MessageBus.BaseTopicPrefix
-
-	payload, err := json.Marshal(data)
+func (svc *Service) Publish(data any, contentType string) error {
+	err := svc.PublishWithTopic(svc.config.Trigger.PublishTopic, data, contentType)
 	if err != nil {
-		return fmt.Errorf("%v: %v", publishMarshalErr, err)
-	}
-	message := types.NewMessageEnvelope(payload, context.Background())
-	err = messageClient.Publish(message, contractsCommon.BuildTopic(baseTopic, triggertopic))
-	if err != nil {
-		return fmt.Errorf("%v: %v", publishDataErr, err)
+		return err
```

Comment:
why not return `fmt.Errorf("%v: %v", publishDataErr, err)`

Reply:
Because PublishWithTopic already does this